libpod/runtime_pod_linux.go

```diff
@@ -210,7 +210,7 @@ func (r *Runtime) removePod(ctx context.Context, p *Pod, removeCtrs, force bool)
 
 		// Ensure state appropriate for removal
 		if err := ctr.checkReadyForRemoval(); err != nil {
-			return errors.Wrapf(err, "pod %s has containers that are not ready to be removed", p.ID())
```

Comment:
I think this doesn't solve the issue - we're going straight through this error message, because all containers *can* be removed. The cause is that we're throwing an error about running containers when the infra is running, instead of throwing an error about the pod having >1 container present.

Reply:
So you think that we should be printing the first of the two errors here? If that is the case then does `.PodContainers()` require some changes to include the exited ones?

Reply:
I don't really know - it's not entirely clear how it's getting this far, I would expect it to be stopped before it enters the loop. Can you reproduce? If so, I would stick some prints in there to figure out what's going on

Reply:
Think I got it, made it so that only running containers are added to the list.